craft_providers/lxd/lxd_instance.py

```diff
@@ -0,0 +1,494 @@
+
+        :returns: State information parsed from lxc if instance exists, else
+            None.
+
+        :raises LXDError: On unexpected error.
+        """
+        instances = self.lxc.list(project=self.project, remote=self.remote)
+
+        # lxc returns a filter instances starting with instance name rather
+        # than the exact instance.  Find the exact match...
+        for instance in instances:
+            if instance["name"] == self.name:
+                return instance
+
+        return None
+
+    def is_mounted(self, *, host_source: pathlib.Path, target: pathlib.Path) -> bool:
+        """Check if path is mounted at target.
+
+        :param host_source: Host path to check.
+        :param target: Instance path to check.
+
+        :returns: True if host_source is mounted at target.
+
+        :raises LXDError: On unexpected error.
+        """
+        disks = self._get_disk_devices()
+
+        return any(
+            disk["path"] == target.as_posix()
+            and disk["source"] == host_source.as_posix()
+            for _, disk in disks.items()
+        )
+
+    def is_running(self) -> bool:
+        """Check if instance is running.
+
+        :returns: True if instance is running.
+
+        :raises LXDError: On unexpected error.
+        """
+        state = self._get_state()
+        if state is None:
+            raise LXDError(brief=f"Instance {self.name!r} does not exist.")
+
+        return state.get("status") == "Running"
+
+    def launch(
+        self,
+        *,
+        image: str,
+        image_remote: str,
+        map_user_uid: bool = False,
+        ephemeral: bool = False,
+    ) -> None:
+        """Launch instance.
+
+        :param image: Image name to launch.
+        :param image_remote: Image remote name.
+        :param uid: Host user ID to map to instance root.
+        :param ephemeral: Flag to enable ephemeral instance.
+
+        :raises LXDError: On unexpected error.
+        """
+        config_keys = dict()
+
+        if map_user_uid:
+            uid = os.getuid()
+            config_keys["raw.idmap"] = f"both {uid!s} 0"
+
+        if self._host_supports_mknod():
+            config_keys["security.syscalls.intercept.mknod"] = "true"
+
+        self.lxc.launch(
+            config_keys=config_keys,
+            ephemeral=ephemeral,
+            instance_name=self.name,
+            image=image,
+            image_remote=image_remote,
+            project=self.project,
+            remote=self.remote,
+        )
+
+    def mount(
+        self,
+        *,
+        host_source: pathlib.Path,
+        target: pathlib.Path,
+        device_name: Optional[str] = None,
+    ) -> None:
+        """Mount host source directory to target mount point.
+
+        Checks first to see if already mounted. If no device name is
+        given, it will be generated with the format:
+         disk_<target.as_posix().replace("/", "_")>
+
+        For example:  /mnt/foo will be named "disk__mnt_foo"
```

Comment:
Can't this create ambiguities between `/mnt/foo` and a mountpoint called `/mnt_foo`?

Reply:
Good point... need to think on that one a bit.

Reply:
Perhaps something similar to what systemd-escape(1) does could help (depending on the set of allowed characters).

Reply:
Actually lxc does not seem to have an issue with any characters, including unicode.  So I'm going to remove the replace.

Reply:
Fixed